### ISSUE TITLE
Potential missing LogIndex after RBS

### ISSUE THREAD

**Issue by ttyusupov** [yugabyte/yugabyte-db#9500]
Jira Link: [DB-1990](https://yugabyte.atlassian.net/browse/DB-1990)
During RBS we don't fetch the Raft log index and then during local bootstrap optimizer can skip log index recreation for some of the first raft log segments, so the Raft log index might be incomplete.

Raft log index is used at least by the following functions:
```
LogReader::LookupOpId
  LogCache::LookupOpId / LogReader::ReadReplicatesInRange
    LogCache::ReadOps 
      PeerMessageQueue::ReadFromLogCache
        PeerMessageQueue::ReadReplicatedMessagesForCDC
        PeerMessageQueue::RequestForPeer
    PeerMessageQueue::IsOpInLog
      PeerMessageQueue::ResponseFromPeer
```

Need to check if it is a legitimate issue or not.

**Comment by bmatican** [comment]
Timur, putting this in your queue for now, as I'm assuming this might go away, once we fix #10960 and can close both then.

**Comment by ttyusupov** [comment]
Fixed by https://github.com/yugabyte/yugabyte-db/issues/10960
